sdk/python/kfp/components/structures.py

```diff
@@ -44,7 +45,9 @@ class InputSpec_(base_model.BaseModel):
         description: Optional: the user description of the input.
     """
     type: Union[str, dict]
-    default: Union[Any, None] = None
+    default: Optional[Any] = None
+    # only used for artifact inputs
+    schema_version: Optional[str] = None
```

Comment:
The artifact class is lost when the component function is converted to `ComponentSpec`, so we need to preserve schema_version to have a full representation of the artifact.